### PR TITLE
Integrate `cardano-ledger` releases

### PR DESCRIPTION
**Pull request by neilmayhew**
Integrate the changes in new versions that `cardano-ledger` recently added to CHaP

### CODE REVIEW

File: ouroboros-consensus/ouroboros-consensus.cabal [hunk 1]
```diff
@@ -278,7 +278,7 @@ library
     bytestring >=0.10 && <0.13,
     cardano-binary,
     cardano-crypto-class,
-    cardano-ledger-core ^>=1.15,
+    cardano-ledger-core >=1.15,
```
Comment (jasagredo) [inline ouroboros-consensus/ouroboros-consensus.cabal:281]: Why not constrain it to `^>=1.16`?

File: ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal [hunk 2]
```diff
@@ -119,7 +119,7 @@ test-suite protocol-test
     base,
     cardano-crypto-class,
     cardano-ledger-binary:testlib,
-    cardano-ledger-core,
+    cardano-ledger-core >=1.16,
```
Comment (jasagredo) [inline ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal:122]: Please use `^>=1.16`.

**Review by jasagredo — APPROVED** [review]
I approve the PR, but please take care of the two comments I left.

File: ouroboros-consensus/ouroboros-consensus.cabal (hunk 1 above)
Comment (lehins) [reply ouroboros-consensus/ouroboros-consensus.cabal]: Done.

File: ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal (hunk 2 above)
Comment (lehins) [reply ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal]: Done

**Pull request merged**